schema/crdb/41.0.0/up05.sql

```diff
@@ -0,0 +1,11 @@
+CREATE TYPE IF NOT EXISTS omicron.public.downstairs_client_stop_reason_type AS ENUM (
+  'replacing',
+  'disabled',
+  'failed_reconcile',
+  'io_error',
+  'bad_negotiation_order',
+  'incompatible',
+  'failed_live_repair',
+  'too_many_outstanding_jobs',
+  'deactivated'
```

Comment:
Do we need a `timeout` choice too?  It could be either a connection closed from the remote side, or just nothing and we gave up on it.

Reply:
I think so, yeah - `ClientStopReason` is recorded now but not if the client task has a ClientRunResult. I'll do that now.

Reply:
df79ff2dc adds separate endpoints for when an Upstairs requests a stop, and when a Downstairs client task stops.